src/dune_pkg/opam_solver.ml

```diff
@@ -482,5 +482,40 @@ let solve_lock_dir solver_env version_preference (repo, repo_id) ~local_packages
       | Ok pkgs_by_name ->
         Lock_dir.create_latest_version pkgs_by_name ~ocaml:None ~repo_id
     in
-    summary, lock_dir)
+    let files =
+      let read path =
+        let rec read acc path =
+          match Path.readdir_unsorted_with_kinds path with
+          | Ok files ->
+            List.concat_map files ~f:(fun (filename, kind) ->
+              match (kind : Unix.file_kind) with
+              | S_REG -> [ acc @ [ filename ] ]
+              | S_DIR -> read (acc @ [ filename ]) (Path.relative path filename)
+              | _ -> [])
+          | Error _ ->
+            (* TODO *)
+            []
+        in
+        read [] path
+        |> List.map ~f:(fun exploded_local_path ->
+          ( Path.L.relative path exploded_local_path
+          , match exploded_local_path with
+            | [] -> assert false
+            | [ filename ] -> Path.Local.of_string filename
+            | dir :: filename -> Path.Local.L.relative (Path.Local.of_string dir) filename
+          ))
+      in
+      match
+        Package_name.Map.of_list_map opam_packages_to_lock ~f:(fun opam_package ->
+          let files_path = Opam_repo.get_opam_package_files_path repo opam_package in
+          ( Package_name.of_string
+              (OpamPackage.Name.to_string (OpamPackage.name opam_package))
+          , read files_path ))
+      with
+      | Ok files -> files
+      | Error _ ->
+        (* What to do here? *)
```

Comment:
Not sure what to do in this situation yet.

Reply:
When is this error possible?

Reply:
Okay, do we have a test for this? If no, could you please add one?

Reply:
There doesn't appear to be a test for it however I am not quite sure how it could be triggered in the first place. The relevant function is `solve_lock_dir` in `opam_solver`. The only way this error can be triggered is if the solver returns duplicate opam packages.